kinesis_manager/src/stream_definition_provider.cpp

```diff
@@ -19,21 +19,27 @@
 #define STREAM_DEFINITION_MAX_CODEC_PRIVATE_DATA_SIZE (1024)
 
 using namespace com::amazonaws::kinesis::video;
+using PathVector = std::vector<std::string>;
 
 namespace Aws {
 namespace Kinesis {
 
+static PathVector Combine(const PathVector & prefix, const char * parameter_name) {
```

Comment:
Instead of defining a new type PathVector, can we make this a feature of appending a path component as part of the PathParameter class itself?

Reply:
The definition of this feature being something like:
```
const ParameterPath ParameterPath::operator+(const std::string & appendee) { ... }
```